jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1EncodedDataRouter.java

```diff
@@ -16,487 +16,421 @@
+
+    ArrayNode encodings = buildEncodings();
+    var embeddedEncodings = xmlMapper.createObjectNode();
+    embeddedEncodings.set(AppContext.ENCODINGS_STRING, encodings);
+
+    metadataNode.set(AppContext.ENCODINGS_STRING, embeddedEncodings);
+
+    ObjectNode message = mapper.createObjectNode();
+    message.set(AppContext.METADATA_STRING, metadataNode);
+    message.set(AppContext.PAYLOAD_STRING, payloadNode);
+
+    ObjectNode root = mapper.createObjectNode();
+    root.set(AppContext.ODE_ASN1_DATA, message);
 
-      return hexEncodedTim;
+    var outputXml = xmlMapper.writeValueAsString(root)
+        .replace("<ObjectNode>", "")
+        .replace("</ObjectNode>", "");
+    log.debug("Fully crafted ASD to be encoded: {}", outputXml);
+    return outputXml;
+  }
+
+  private ArrayNode buildEncodings() throws JsonUtilsException {
+    ArrayNode encodings = mapper.createArrayNode();
+    encodings.add(TimTransmogrifier.buildEncodingNode(ADVISORY_SITUATION_DATA_STRING,
+        ADVISORY_SITUATION_DATA_STRING,
+        EncodingRule.UPER));
+    return encodings;
+  }
 
-    } catch (JsonUtilsException e1) {
-      log.error("Unable to parse signed message response ", e1);
+  private void sendToRsus(ServiceRequest request, String encodedMsg) {
+    if (null == request.getSnmp() || null == request.getRsus()) {
+      log.debug("No RSUs or SNMP provided. Not sending to RSUs.");
+      return;
     }
-    return encodedTIM;
+    log.info("Sending message to RSUs...");
+    rsuDepositor.deposit(request, encodedMsg);
   }
 
-  /**
-   * Checks if header is present in encoded message.
-   */
-  private boolean isHeaderPresent(String encodedTim) {
-    return encodedTim.indexOf("001F") > 0;
+  private void setRequiredExpiryDate(SimpleDateFormat dateFormat, String timStartDateTime,
+                                     int maxDurationTime, JSONObject timWithExpiration) {
+    try {
+      Date timTimestamp = dateFormat.parse(timStartDateTime);
+      Date requiredExpirationDate = new Date();
+      requiredExpirationDate.setTime(timTimestamp.getTime() + maxDurationTime);
+      timWithExpiration.put("requiredExpirationDate", dateFormat.format(requiredExpirationDate));
+    } catch (Exception e) {
+      log.error("Unable to parse requiredExpirationDate. Setting requiredExpirationDate to 'null'", e);
+      timWithExpiration.put("requiredExpirationDate", "null");
+    }
+  }
+
+  private void setExpiryDate(SignatureResultModel signedResponse,
+                             JSONObject timWithExpiration,
+                             SimpleDateFormat dateFormat) {
+    try {
+      timWithExpiration.put("expirationDate",
+          dateFormat.format(new Date(signedResponse.getResult().getMessageExpiry() * 1000))
+      );
+    } catch (Exception e) {
+      log.error("Unable to get expiration date from signed messages response. Setting expirationData to 'null'", e);
+      timWithExpiration.put("expirationDate", "null");
+    }
```

Comment:
Note: this is one of the rare times in this code where we can safely continue (as far as I can tell from the code) processing a message should an Exception be thrown. I intentionally handled the expectation here and allowed the processing to continue. I couldn't find any downstream issues in the code. Also, please note that we already handled the exception and continued processing in the previous implementation. This does not change the code flow.